apps/studio/src/components/ResourceSelector/ResourceItem.tsx

```diff
@@ -0,0 +1,109 @@
+import type { IconType } from "react-icons"
+import { Suspense, useMemo } from "react"
+import { Box, HStack, Icon, Mark, Skeleton, Text } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+import { QueryErrorResetBoundary } from "@tanstack/react-query"
+import fuzzysort from "fuzzysort"
+import { ErrorBoundary } from "react-error-boundary"
+import { BiData, BiFile, BiFolder, BiHome, BiLockAlt } from "react-icons/bi"
+
+import type { RouterOutput } from "~/utils/trpc"
+
+type ResourceItemProps = Pick<
+  RouterOutput["resource"]["getChildrenOf"][number],
+  "permalink" | "type"
+> & {
+  isSelected: boolean
+  isDisabled: boolean
+  searchQuery: string
+  onResourceItemSelect: () => void
+}
+
+const SuspendableResourceItem = ({
+  permalink,
+  type,
+  isSelected,
+  isDisabled,
+  searchQuery,
+  onResourceItemSelect,
+}: ResourceItemProps) => {
+  const icon: IconType = useMemo(() => {
+    switch (type) {
+      case "Folder":
+        return BiFolder
+      case "CollectionPage":
+      case "Page":
+        return BiFile
+      case "Collection":
+        return BiData
+    }
+  }, [type])
+
+  const SearchTextHighlight = useMemo(() => {
+    const result = fuzzysort.single(searchQuery, permalink)
+    if (!result) {
+      return permalink
+    }
+
+    return fuzzysort.highlight(result, (m, i) => (
+      <Mark key={i} bgColor="interaction.tinted.main.active">
+        {m}
+      </Mark>
+    ))
+  }, [searchQuery, permalink])
+
+  return (
+    <Button
+      variant="clear"
+      w="full"
+      justifyContent="flex-start"
+      color={isSelected ? "interaction.main.default" : "base.content.default"}
+      bgColor={isSelected ? "interaction.muted.main.active" : undefined}
+      isDisabled={isDisabled}
+      _active={{
+        bgColor: !isDisabled ? "interaction.tinted.main.active" : undefined,
+      }}
+      _disabled={{ color: "interaction.support.disabled-content" }}
+      _hover={{
+        bgColor:
+          !isSelected && !isDisabled
+            ? "interaction.muted.main.hover"
+            : undefined,
+      }}
```

Comment:
supernit: 

actually you can set `selected` styling with `_selected` prop too. Just need the `data-selected` attribute. 

Might be easier to parse. disabled state too.

```suggestion
      color="base.content.default"
      isDisabled={isDisabled}
      data-selected={dataAttr(isSelected)} // custom function from chakra-ui
      _selected={{
        color: "interaction.main.default",
        bg: "interaction.muted.main.active"
      }}
      _disabled={{ 
        color: "interaction.support.disabled-content",
        _active: {{
          bg: "interaction.support.disabled"
        }}
        _hover: {{
          bg: "interaction.support.disabled"
        }}
      }}
      _hover={{
        bg: "interaction.muted.main.hover"
      }}
```

Reply:
Ah TIL about the `data-selected` prop, updated in 7f139851126b5ab4f72d99fbbf9c5588cb2f9adc.